cli/internal/helm/loader.go

```diff
@@ -87,35 +93,39 @@ func NewLoader(csp cloudprovider.Provider, k8sVersion versions.ValidK8sVersion,
 	case cloudprovider.OpenStack:
 		ccmImage = versions.VersionConfigs[k8sVersion].CloudControllerManagerImageOpenStack
 	}
-
-	// TODO(malt3): Allow overriding container image registry + prefix for all images
-	// (e.g. for air-gapped environments).
-	return &ChartLoader{
+	return &chartLoader{
+		cliVersion:                   cliVersion,
 		csp:                          csp,
-		joinServiceImage:             imageversion.JoinService("", ""),
-		keyServiceImage:              imageversion.KeyService("", ""),
+		idFile:                       idFile,
 		ccmImage:                     ccmImage,
 		azureCNMImage:                cnmImage,
+		config:                       config,
+		joinServiceImage:             imageversion.JoinService("", ""),
+		keyServiceImage:              imageversion.KeyService("", ""),
 		autoscalerImage:              versions.VersionConfigs[k8sVersion].ClusterAutoscalerImage,
 		verificationServiceImage:     imageversion.VerificationService("", ""),
 		gcpGuestAgentImage:           versions.GcpGuestImage,
 		konnectivityImage:            versions.KonnectivityAgentImage,
 		constellationOperatorImage:   imageversion.ConstellationNodeOperator("", ""),
 		nodeMaintenanceOperatorImage: versions.NodeMaintenanceOperatorImage,
-		clusterName:                  clusterName,
 	}
 }
 
+// ReleaseApplyOrder is a list of releases in the order they should be applied.
+// makes sure if a release was removed as a dependency from one chart,
+// and then added as a new standalone chart (or as a dependency of another chart),
+// that the new release is installed after the existing one to avoid name conflicts.
+type ReleaseApplyOrder []Release
+
 // LoadReleases loads the embedded helm charts and returns them as a HelmReleases object.
-func (i *ChartLoader) LoadReleases(
-	config *config.Config, conformanceMode bool, helmWaitMode WaitMode, masterSecret uri.MasterSecret,
-	serviceAccURI string, idFile clusterid.File, output terraform.ApplyOutput,
-) (*Releases, error) {
+func (i *chartLoader) LoadReleases(conformanceMode bool, helmWaitMode WaitMode, masterSecret uri.MasterSecret,
```

Comment:
since the chartLoader is already private there is no point in making `LoadReleases` private is there? I think it makes it more intuitive that this is the exposing function of the struct.

Reply:
If any function of the `helm` package returns the (private) `chartLoader`, the caller of that function will be able to call `LoadReleases` on the returned struct.
I feel like exported functions are less likely to be called from the package they are defined in, so this (imo) does not help much with showing a developer what method of `chartLoader` should be called from other parts of the `helm` package.